include/TrustWalletCore/TWBitsharesAddress.h

```diff
@@ -0,0 +1,49 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "TWBase.h"
+#include "TWData.h"
+#include "TWString.h"
+
+
+TW_EXTERN_C_BEGIN
+
+struct TWPublicKey;
+
+/// Represents a Bravo address.
+TW_EXPORT_CLASS
+struct TWBitsharesAddress;
+
+/// Compares two addresses for equality.
+TW_EXPORT_STATIC_METHOD
+bool TWBitsharesAddressEqual(struct TWBitsharesAddress *_Nonnull lhs, struct TWBitsharesAddress *_Nonnull rhs);
+
+/// Determines if the string is a valid address.
+TW_EXPORT_STATIC_METHOD
+bool TWBitsharesAddressIsValidString(TWString *_Nonnull string);
+
+/// Creates an address from a string representaion.
+TW_EXPORT_STATIC_METHOD
+struct TWBitsharesAddress *_Nullable TWBitsharesAddressCreateWithString(TWString *_Nonnull string);
+
+/// Creates an address from a public key.
+TW_EXPORT_STATIC_METHOD
+struct TWBitsharesAddress *_Nonnull TWBitsharesAddressCreateWithPublicKey(struct TWPublicKey *_Nonnull publicKey);
+
+/// Creates an address from a key hash.
+TW_EXPORT_STATIC_METHOD
+struct TWBitsharesAddress *_Nullable TWBitsharesAddressCreateWithKeyHash(TWData *_Nonnull keyHash);
```

Comment:
where do we need to create address from keyHash?

Reply:
Sorry, that was supposed to be create from raw data. Should we rename or entirely remove the function?

Reply:
just remove it, in general we don't need to create address from key hash